source/material_model/rheology/strain_dependent.cc

```diff
@@ -199,20 +199,23 @@ namespace aspect
             const unsigned int n_s12 = this->introspection().compositional_index_for_name("s12");
             const unsigned int n_s21 = this->introspection().compositional_index_for_name("s21");
             const unsigned int n_s22 = this->introspection().compositional_index_for_name("s22");
-            AssertThrow(n_s12>n_s11 && n_s21>n_s12 && n_s22 > n_s21,
-                        ExcMessage("Material model visco_plastic with strain weakening using the full strain tensor only works if there "
-                                   "are compositional fields called sij, with i=1,..,dim and j=1,...,dim in the order s11,s12,s21 etc."));
+            if (dim==2)
+              {
+                AssertThrow(n_s12>n_s11 && n_s21>n_s12 && n_s22 > n_s21,
```

Comment:
I would find it easier to read if you write ``s11 < s12 && s12 < s21 ...``

Reply:
agree :+1: